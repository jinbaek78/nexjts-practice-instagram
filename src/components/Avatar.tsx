import Image from 'next/image';

type Props = {
  src: string;
  rainbow?: boolean;
  width?: number;
};
const IMAGE_CONTAINER_CLASS =
  'rounded-full bg-gradient-to-r from-pink-500 via-red-500 to-yellow-500 p-1 w-full';
export default function Avatar({ src, rainbow = false, width = 100 }: Props) {
  return (
    <>
      <div
        className={rainbow ? IMAGE_CONTAINER_CLASS : 'rounded-full'}
        style={{
          borderRadius: '999px',
          overflow: 'hidden',
          width: `${width}px`,
          height: `${width}px`,
          objectFit: 'cover',
        }}
      >
        <Image
          className="rounded-full bg-white p-1 w-full h-full"
          src={src || ''}
          alt="avatar"
          width={width}
          height={100}
        />
      </div>
    </>
  );
}
