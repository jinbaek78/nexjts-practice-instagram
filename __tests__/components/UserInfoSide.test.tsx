import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { useSession } from 'next-auth/react';
import { fakeSession } from '@/tests/mock/session';
import Avatar from '@/components/Avatar';
import UserInfoSide from '@/components/UserInfoSide';

jest.mock('@/components/Avatar');
jest.mock('next-auth/react', () => ({
  useSession: jest.fn(),
}));

describe('UserInfoSide', () => {
  const { name, email, image } = fakeSession!.user!;

  afterEach(() => {
    (useSession as jest.Mock).mockReset();
    (Avatar as jest.Mock).mockReset();
  });

  it('renders', () => {
    (useSession as jest.Mock).mockImplementation(() => ({ data: fakeSession }));
    const { container } = render(<UserInfoSide />);
    expect(container).toMatchSnapshot();
  });
  it('renders with user information', () => {
    (useSession as jest.Mock).mockImplementation(() => ({ data: fakeSession }));
    render(<UserInfoSide />);

    expect(Avatar).toHaveBeenCalledTimes(1);
    expect((Avatar as jest.Mock).mock.calls[0][0].src).toBe(image);

    expect(screen.getByText(name!)).toBeInTheDocument();
    expect(screen.getByText(email!.split('@')[0])).toBeInTheDocument();
  });
});
