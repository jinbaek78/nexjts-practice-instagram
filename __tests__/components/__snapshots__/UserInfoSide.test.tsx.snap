// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoSide renders 1`] = `
<div>
  <section
    class="flex flex-col gap-10"
  >
    <div
      class="flex items-center"
    >
      <div
        class="ml-5"
      >
        <h3
          class="text-xl font-bold"
        >
          abded
        </h3>
        <h3
          class="text-2xl text-zinc-600"
        >
          jin
        </h3>
      </div>
    </div>
    <p
      class="text-zinc-600 text-lg"
    >
      About﹒ Help﹒Press﹒API﹒Jobs﹒
      <br />
      Privacy﹒Terms﹒Location﹒
      <br />
      Language
    </p>
    <p
      class="text-zinc-700 font-semibold"
    >
      Copyright INSTANTGARM
      <br />
      from MENTAL
    </p>
  </section>
</div>
`;
